examples/federated-learning/players.py

```diff
@@ -130,6 +134,18 @@ def _update_one_round(self, data_owners, **kwargs):
       self.optimizer.apply_gradients(zip(aggr_gradients,
                                          self.model.trainable_variables))
 
+  def save_model(self, save_dir):
+    if not os.path.exists(save_dir):
+      os.makedirs(save_dir)
```

Comment:
can change this block to `os.path.makedirs(save_dir, exist_ok)` to avoid a race condition between the calls to `exists` and `makedirs`